### PR TITLE
working on NetBSD support.

### PR DESCRIPTION
**Pull request by epochqwert**

### CODE REVIEW

File: client/AdminClient.c [hunk 1]
```diff
@@ -119,11 +119,14 @@ static Iface_DEFUN receiveMessage(struct Message* msg, struct Iface* addrIface)
 
     struct Sockaddr_storage source;
     Message_pop(msg, &source, ctx->targetAddr->addrLen, NULL);
-    if (Bits_memcmp(&source, ctx->targetAddr, ctx->targetAddr->addrLen)) {
+    if (Bits_memcmp(&source+1, ctx->targetAddr+1, ctx->targetAddr->addrLen-1)) {
```
Comment (kpcyrd) [inline client/AdminClient.c]: I don't think this is a good idea...

File: interface/tuntap/TUNInterface_netbsd.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                                   struct Except* eh,
+                                   struct Allocator* alloc)
+{
+    if (isTapMode) { Except_throw(eh, "tap mode not supported on this platform"); }
+
+    int i;
+    // to store the tunnel device index
+    int ppa = 9;
+    // Open the descriptor
+    int tunFd = open("/dev/tun9", O_RDWR);
```
Comment (Kubuxu) [inline interface/tuntap/TUNInterface_netbsd.c]: Why `/dev/tun9/`?

File: client/AdminClient.c (hunk 1 above)
Comment (cjdelisle) [inline client/AdminClient.c]: You're skipping the wrong byte and you're doing it for all platforms, including those which do not have it. This byte is also in OSX and FreeBSD so you need to find where it is being properly cleared for those platforms (it is ok to clear the byte, it does no harm for it to be zero)

File: interface/tuntap/TUNInterface_netbsd.c (hunk 2 above)
Comment (cjdelisle) [inline interface/tuntap/TUNInterface_netbsd.c]: The correct way is to loop over tun devices until you find one which opens properly. This is how we open the BPF device on Darwin: https://github.com/cjdelisle/cjdns/blob/master/interface/ETHInterface_darwin.c#L230

**Comment by cjdelisle** [comment]
open a new one when you're ready

**Comment by epochqwert** [comment]
Yeah. Good idea.

**Pull request closed without merging**